lib/services/nativescript-project-plugins-service.ts

```diff
@@ -188,25 +191,47 @@ export class NativeScriptProjectPluginsService implements IPluginsService {
 
 	public fetch(pluginIdentifier: string): IFuture<void> {
 		return (() => {
+			let pluginBasicInfo = this.fetchPluginBasicInformation(pluginIdentifier, "fetch").wait();
+			this.$logger.printMarkdown(util.format("Successfully fetched plugin `%s`.", pluginBasicInfo.name));
+		}).future<void>()();
+	}
+
+	public fetchPluginBasicInformation(pluginIdentifier: string, failMessageMethodName: string,  opts?: { isTgz: boolean }): IFuture<IBasicPluginInformation> {
+		return ((): IBasicPluginInformation => {
 			if(!pluginIdentifier) {
 				this.$errors.fail("You must specify local path, URL to a plugin repository, name or keywords of a plugin published to the NPM.");
 			}
+			let pluginIdentifierExtname = path.extname(pluginIdentifier),
+				isTgz = opts && opts.isTgz || pluginIdentifierExtname === ".tgz" || pluginIdentifierExtname === ".gz";
+
+			if (isTgz) {
+				pluginIdentifier = path.resolve(pluginIdentifier);
+			}
 
-			let pathToInstalledPlugin = this.installPackageToTempDir(pluginIdentifier).wait();
-			if(pathToInstalledPlugin) {
+			let pathToInstalledPlugin = this.installPackageToTempDir(pluginIdentifier).wait(),
+				actualPlugin: string,
+				installLocalPluginOptions: any = { addPluginToPackageJson: true };
+
+			if (isTgz) {
+				actualPlugin = pluginIdentifier;
+				installLocalPluginOptions["packageJsonContents"] = this.$fs.readJson(path.join(pathToInstalledPlugin, this.$projectConstants.PACKAGE_JSON_NAME)).wait();
+			} else {
+				actualPlugin = pathToInstalledPlugin;
+			}
+
+			if (pathToInstalledPlugin) {
 				let pluginsPath = path.join(this.$project.getProjectDir().wait(), "plugins");
 				// use cp instead of mv, as it would fail if pathToInstalledPlugin is mounted
 				// on a different device from the pluginsPath with error:
 				// Error: EXDEV, cross-device link not permitted
-				shelljs.cp("-Rf", [pathToInstalledPlugin], pluginsPath);
-				let pluginBasicInfo = this.installLocalPlugin(path.join(pluginsPath, path.basename(pathToInstalledPlugin)), {addPluginToPackageJson: true}).wait();
-				this.$logger.printMarkdown(util.format("Successfully fetched plugin `%s`.", pluginBasicInfo.name));
+				shelljs.cp("-Rf", [actualPlugin], pluginsPath);
+				return this.installLocalPlugin(path.join(pluginsPath, path.basename(actualPlugin)), installLocalPluginOptions).wait();
```

Comment:
It is actually defined as a property of `installLocalPluginOptions` a little above